sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -286,13 +288,13 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
       ResolveFieldNameAndPosition ::
       AddMetadataColumns ::
       DeduplicateRelations ::
-      ResolveReferences ::
+      new ResolveReferences(v1SessionCatalog) ::
```

Comment:
optional: you can make these `case class`es instead, to avoid the `new` keyword here and below

Reply:
Addressed